service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
+1 to what's currently in the commit -- I use the same style in the axiom-api project as well since it seemed like that's what was recommended by the Immutables project